fvm/src/machine/engine.rs

```diff
@@ -105,13 +114,41 @@ pub fn default_wasmtime_config() -> wasmtime::Config {
     // > not enabled by default.
     c.cranelift_nan_canonicalization(true);
 
+    // wasmtime default: 512k
     // Set to something much higher than the instrumented limiter.
-    c.max_wasm_stack(64 << 20).unwrap();
+    // Note: This is in bytes, while the instrumented limit is in stack elements
+    c.max_wasm_stack(4 << 20).unwrap();
 
     // Execution cost accouting is done through wasm instrumentation,
     c.consume_fuel(false);
-
-    // c.cranelift_opt_level(Speed); ?
+    c.epoch_interruption(false);
+
+    // Disable debug-related things, wasm-instrument doesn't fix debug info
+    // yet, so those aren't useful, just add overhead
+    c.debug_info(false);
+    c.generate_address_map(false);
+    c.cranelift_debug_verifier(false);
+
+    // Reiterate some defaults
+    c.guard_before_linear_memory(true);
+    c.interruptable(false);
+    c.parallel_compilation(true);
+
+    c.allocation_strategy(InstanceAllocationStrategy::Pooling {
+        strategy: PoolingAllocationStrategy::NextAvailable,
+        instance_limits: InstanceLimits{
+            count: 1000, // todo set this higher than recursive call limit?
```

Comment:
Only real todo here, I think.

Reply:
What happens when an instance is requested and the pool is exhausted? Is one created on demand, or does wasm instantiation fail?

Reply:
`Instance::new()` will return an error

> Err: Module/instance resource limits are exceeded.

https://docs.rs/wasmtime/latest/wasmtime/struct.Instance.html#impl

Reply:
Hmmm, I don't think we want this then. We either have to allocate 4096 instances upfront (call stack depth limit), which is wasteful because most messages won't even reach a few instances, or use the on-demand strategy.

@magik6k if you agree, could you please revert this change? I'd be happy with a hybrid model (once we actually characterise its footprint), but that doesn't seem to exist yet.